Add a job for generating API, CLI, and related test reports

JacobCallahan opened the pull request:
This job should be ran along with each automation run.

JacobCallahan commented:
This is my initial draft, as I'm sure I've done a number of things incorrectly.
Any advice would be greatly appreciated!

JacobCallahan commented on jobs/api-cli-reports.yaml:
```diff
@@ -0,0 +1,59 @@
+            name: ROBOTTELO_BRANCH
+            default: master
+            description: You can override this to any branch. Example - 6.4.z
+    scm:
```
@san7ket @ntkathole I'm pretty sure this scm section is incorrect..

san7ket replied:
yea , correct this will probably overwrite repos in workspace. 
but this is possible with pipeline SCM step, where you can clone multiple git repos in separate target dir's.

san7ket commented on jobs/api-cli-reports.yaml:
```diff
@@ -0,0 +1,59 @@
+- job:
```
Job should be a pipeline job

ogajduse replied:
Exactly, this was the first thing that came to my mind when I saw the code.

san7ket commented on jobs/satellite6-automation/satellite6-projects.yaml:
```diff
@@ -39,6 +39,7 @@
         - 'report-automation-results-{satellite_version}-{os}'
         - 'report-consolidated-coverage-{satellite_version}-{os}'
         - 'automation-{satellite_version}-trigger-tiers-{os}'
+        - 'api-cli-reports'
```
This won't work as the job is not part of project matrix, should be treated as standalone job. 
We can safely remove it from here

san7ket commented on scripts/api-cli-reports.sh:
```diff
@@ -0,0 +1,36 @@
+#! /bin/bash
```
Convert to groovy. 
We are supposed to be using pipeline jobs and avoid adding new freestyle jobs. 
We are mostly using declarative pipelines syntaxes.

san7ket commented on jobs/api-cli-reports.yaml:
```diff
@@ -0,0 +1,59 @@
+    wrappers:
+        - build-name:
+            name: '#${BUILD_NUMBER} API/CLI Reports'
+        - build-user-vars
```
I am unable find this wrapper.

san7ket commented on jobs/api-cli-reports.yaml:
```diff
@@ -0,0 +1,59 @@
+            name: '#${BUILD_NUMBER} API/CLI Reports'
+        - build-user-vars
+        - default-wrappers
+    builders:
```
With Pipeline you won't be able to use shining panda to get virtual env. But worry not, @lpramuk has already created a shared library to have the py virtual env in jobs. look into robottelo-ci/vars dir

JacobCallahan commented:
@san7ket @ntkathole I've done an update to the new format. Again, I'm sure I got some things wrong with this!
Also, Travis is giving errors about the AUTO_TOOLS_REPO parameter being missing. That *shouldn't* be anything I touched, but if so I can correct it.
Thanks

san7ket commented:
@JacobCallahan the failure is due to new JJB regression. We have pined JJB , will restart job

JacobCallahan commented on jobs/api-cli-reports.yaml:
```diff
@@ -0,0 +1,36 @@
+    pipeline-scm:
+        scm:
+          - git:
+                url: https://github.com/JacobCallahan/plinko.git
```
ahhh ok. I misunderstood this line!

JacobCallahan commented on workflows/qe/api-cli-reports.groovy:
```diff
@@ -0,0 +1,106 @@
+            steps {
+                sh_venv '''
+                    apix makelib -n satellite -l $(SATELLITE_VERSION) -t advanced --data-dir $(DATA_DIR)
+                    cd ..
```
apix, clix, and plinko should all have their own directories relative to the base. I'm not sure what the entry point of each ssh_venv command is, so I may have to handle directory traversal more or less.

ntkathole replied:
Jenkins pipeline syntax provides 
```
dir("folder") {
sh_venv  '''
    pwd
'''
}
```
You can use this

JacobCallahan commented on workflows/qe/api-cli-reports.groovy:
```diff
@@ -0,0 +1,106 @@
+                sh_venv """
+                    export PYCURL_SSL_LIBRARY=\$(curl -V | sed -n 's/.*\\(NSS\\|OpenSSL\\).*/\\L\\1/p')
+                    pip install -r requirements.txt
+                    pip install .
```
I believe it needs to be included as robottelo needs to be installed as a package for plinko to import it as a module.

san7ket replied:
Same here , can you check the code on staging first. 
I assume, pip install . will be run in root of workspace , but not inside the robottelo dir

JacobCallahan commented on workflows/qe/api-cli-reports.groovy:
```diff
@@ -0,0 +1,106 @@
+            archiveArtifacts(artifacts: '*.yaml', allowEmptyArchive: true)
+        }
+    }
+}
```
will do

JacobCallahan commented on jobs/api-cli-reports.yaml:
```diff
@@ -0,0 +1,36 @@
+        lightweight-checkout: true
+    wrappers:
+        - default-wrappers
+        - config-file-provider:
```
Here's a question: Is that needed for my groovy file? I'm not pulling anything from the configs, unless the git defaults require it. Am I mistaken on this?

ntkathole replied:
@JacobCallahan you can remove it completely, if you don't need anything from gitlab jenkins-config.

san7ket commented on workflows/qe/api-cli-reports.groovy:
```diff
@@ -0,0 +1,106 @@
+            steps {
+                git defaults.apix
+                sh_venv """
+                    git clone https://github.com/JacobCallahan/apix.git
```
No need for this as git defaults.apix will clone the repo.

san7ket commented on workflows/qe/api-cli-reports.groovy:
```diff
@@ -0,0 +1,106 @@
+            steps {
+                git defaults.plinko
+                sh_venv """
+                    pip install .
```
This will not be in plinko dir , you need to cd in first

JacobCallahan commented:
@san7ket @ntkathole I think I've addressed all the comments I could. There are still some pending answers above if you're able to answer them. Namely, those are the need for a config file reference and the originating path of each ssh_env.

JacobCallahan commented:
@san7ket @ntkathole Alright. I believe everything has been addressed with the version I just pushed up. I can squash everything if this looks good to you both.

JacobCallahan commented:
@san7ket @ntkathole any additional reviews?

JacobCallahan commented:
@san7ket @ntkathole i'm running into too many local issues to get this added to stage. If one of you is bale, that would help. Otherwise, we can merge. Deadline for the merge is this Friday (9th).
